lib/transaction.js

```diff
@@ -207,22 +192,9 @@ class Transaction extends EventEmitter {
   // Acquire a connection and create a disposer - either using the one passed
   // via config or getting one off the client. The disposer will be called once
   // the original promise is marked completed.
-  async acquireConnection(config, cb) {
-    const configConnection = config && config.connection;
-    const connection =
-      configConnection || (await this.client.acquireConnection());
-
-    try {
-      connection.__knexTxId = this.txid;
-      return await cb(connection);
-    } finally {
-      if (!configConnection) {
-        debug('%s: releasing connection', this.txid);
-        this.client.releaseConnection(connection);
-      } else {
-        debug('%s: not releasing external connection', this.txid);
-      }
-    }
+  async acquireConnection(connection, cb) {
```

Comment:
This method probably should be renamed, because its not about acquiring connection anymore

Reply:
Yeah -- I agree that the method should be renamed.  The question is whether or not we want to maintain backwards-compatibility.  In particular: it appears that several developers have worked around limitations in `knex` by calling `acquireConnection(..)` directly.  So, renaming the method would break all of those work-arounds.

At the moment, I'm continuing w/ some refactoring using the current names.  At the end, I might create some method aliases / etc. so that we can maintain as much backwards-compatibility as possible.  (And, probably have the aliases log a deprecation warning of some kind)